addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -84,18 +93,95 @@ def _default_holiday_status_id(self):
     validation_type = fields.Selection('Validation Type', related='holiday_status_id.validation_type')
     can_reset = fields.Boolean('Can reset', compute='_compute_can_reset')
     can_approve = fields.Boolean('Can Approve', compute='_compute_can_approve')
+    accrual = fields.Boolean("Accrual", related='holiday_status_id.accrual', store=True, readonly=True)
```

Comment:
I think we should add a onchange on `holiday_status_id` to reset `number_of_days_temp` to zero if the selected type is accrual, otherwise, it is possible to create an allocation with an accrual leave type with some days already allocated.
And the opposite, when not accrual, reset interval number/unit fields.